tests/acceptance/network/test_middleware_retries.py

```diff
@@ -0,0 +1,32 @@
+from unittest.mock import Mock
+
+import maya
+from web3.types import RPCEndpoint
+
+from nucypher.blockchain.middleware.retry import RetryRequestMiddleware
+from tests.constants import RPC_TOO_MANY_REQUESTS
+
+
+# TODO - since this test does exponential backoff it takes >= 2^1 = 2s, should we only run on CI?
+def test_request_with_retry_exponential_backoff():
```

Comment:
👍 - nice to move this slower test to acceptance.

Reply:
It's a bit out-of-scope in this PR, but we snuck it in, considering it was responsible for ~80% of the unit test runtime.